worker/upload_test.go

```diff
@@ -0,0 +1,276 @@
+package worker
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"sync"
+	"testing"
+	"time"
+
+	"go.sia.tech/core/types"
+	"go.sia.tech/renterd/api"
+	"go.sia.tech/renterd/object"
+	"go.uber.org/zap"
+	"lukechampine.com/frand"
+)
+
+type (
+	mockMemory        struct{}
+	mockMemoryManager struct{}
+	mockObjectStore   struct {
```

Comment:
It might be bit easier to mock the bus if we reduce the size of the object store interface by getting rid of `UpdateObject`, `UpdateSlab` and all the other functions that don't necessarily need to be in the upload manager.

Reply:
That's something I'm looking into already indeed, the upload and download manager need only 3 methods. I'm having trouble with the naming. I actually had it at one point but I had to call it an `Uploader interface` and didn't want to conflict with the uploader type we already have.

I'll do this in a F/U if that's ok.